### DISCUSSION TITLE
ترجمة أدوات واجهة التفاعلية مثل computed, watch و ref ...

### DISCUSSION THREAD

**Discussion by boussadjra** [vuejs-translations/docs-ar#136] [مصطلحات] [answered]
[اتفقنا بشكل مبدئي على ترجمة Reactivity API بمصطلح "واجهة التفاعلية"](https://github.com/vuejs-translations/docs-ar/discussions/135) والتي توفر بعض الأدوات لإنشاء حالة تفاعلية، وهنا سأسرد ترجمات لأشهرها.
-  الكائن المُعاد من دالة ref يسمى مرجع تفاعلي، فهنا مثلا `const count = ref(0)` نقول أن count هو مرجع تفاعلي. 
- الكائن المعاد من دالة reactive يسمى كائن تفاعلي أو وسيط تفاعلي
- الكائن المعاد من دالة computed يمكن أن يشار إليه عليه بمرجع محسوب أو إذا أردنا أن نشمل بذلك واجهة الخيارات فنقول خاصية محسوبة، و اعتقد أن عبارة خاصية محسوبة أسهل وأشمل في المعنى
- دالة watch يمكن ترجمتها لمُراقب أو خاصية مُراقبة.

**Comment by Abdelaziz18003** [comment, marked as answer]
شكرا أخي إبراهيم علي هذه المناقشة المهمة، أعتذر عن التأخر في التعقيب، فلم أرد التعليق قبل أن أرجع إلى الصفحات الأصلية وأقرأها بعناية حتى أفهم جيدا المعنى الأصلي للمصطلحات. إليك ما انتهيت إليه في هذه المسألة

- الكائن المعاد من الدالة reactive: أرى أن "كائن تفاعلي" التي ذكرتها تفي بالغرض وتعبر عن المعنى، وأرى الإقتصار عليها في الإستعمال العادي او للإشارة للقيمة المعادة من الدالة، أما عند الحديث التفصيلي عن نوع الكائن المعاد أو عند الإشارة إلى Reactive Proxy فأرى أن كلمة "وكيل" هي أقرب ل "proxy" من "وسيط" وهي الكلمة المنتشرة لتلك الكلمة [صفحة ويكيبيديا](https://ar.wikipedia.org/wiki/%D9%88%D9%83%D9%8A%D9%84_(%D8%AD%D9%88%D8%B3%D8%A8%D8%A9))
- أتفق معك على "خاصية محسوبة" فهي تصلح لكلا الواجهتين
- بالنسبة ل watch: فأرى أن نميز بين watch التي يمكن ترجمتها ب"الدالة المراقِبة" و "watched property" التي أتفق معك على ترجتمتها إلى  "خاصية مراقبَة"
- أما كلمة ref فهي اختصار لكلمة reference وهي تأتي على معنيين شائعين هما: مرجع وإشارة والفرق بينهما في المعنى فرق طفيف، فمثلا في الأبحاث الأكاديمية توضع المراجع في آخر البحث، وتوضع إشارات لها داخل البحث، المرجع هو الكتاب الأصلي الذي أخذت منه المعلومة أما الرقم الذي يقودنا إليه فيسمى "رقم الإشارة/الإحالة" بالإسقاط على سياق التفاعلية هنا يكون المرجع حقيقة هو المتغير الذي يُرجَع إليه في النهاية أي المتغير الأصلي أما القيمة المعادة من الدالة ref فأرى أن المعنى المقصود بها هنا هو معنى الإشارة والله أعلم، لذا أقترح "إشارة تفاعلية" بدلا من "مرجع تفاعلي"

**Reply by boussadjra** [reply]
اتفق معك، وأرى أن نعتمد هاته الترجمات.

**Reply by Abdelaziz18003** [reply]
حسنا، بما أننا متفقين على المصطلحات التي وردت في التعليق، سأختاره كجواب لهذه المناقشة
